### PR TITLE
Added cli doc generation and refactored examples 

### PR DESCRIPTION
**Pull request by zr-msft**
…Examples field

Here is a `launch.json` for testing:

```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Generate Docs",
            "type": "go",
            "request": "launch",
            "mode": "auto",
            "program": "${fileDirname}",
            "env": 
                {
                    "GEN_CLI_DOCS": "true"
                },
        }
    ]
}
```

MD files are generated in `notation-reference-doc` but have been added to `.gitignore` to avoid adding them to the repo

If environment variable is absent, behavior should be normal.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -18,15 +18,18 @@ require (
...
 	github.com/golang-jwt/jwt/v4 v4.5.0 // indirect
 	github.com/inconshreveable/mousetrap v1.1.0 // indirect
+	github.com/russross/blackfriday/v2 v2.1.0 // indirect
 	github.com/veraison/go-cose v1.0.0 // indirect
 	github.com/x448/float16 v0.8.4 // indirect
 	golang.org/x/crypto v0.9.0 // indirect
 	golang.org/x/mod v0.10.0 // indirect
 	golang.org/x/sync v0.2.0 // indirect
 	golang.org/x/sys v0.8.0 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
```
Comment (priteshbandi) [inline go.mod:34]: were this new dependencies added automatically when you ran `go mod tidy` ?

Comment (zr-msft) [reply go.mod:34]: @priteshbandi yes

based on some digging, it came from adding this:
https://github.com/notaryproject/notation/pull/687/files#diff-44385ed93d507652b8fe6326f87be189cf97111ba62a17b145bc415752b83daaR12

The `cobra` `go.mod` requires the `go-md2man` and `yaml.v3`:
https://github.com/spf13/cobra/blob/main/go.mod#L6-L9

The `go-md2man` `go.mod` requires `blackfriday`:
https://github.com/cpuguy83/go-md2man/blob/master/go.mod#L5

File: cmd/notation/policy/import.go [hunk 2]
```diff
@@ -24,11 +24,9 @@ func importCmd() *cobra.Command {
 		Short: "Import trust policy configuration from a JSON file",
 		Long: `Import trust policy configuration from a JSON file.
 
-** This command is in preview and under development. **
-
-Example - Import trust policy configuration from a file:
-  notation policy import my_policy.json
-`,
+** This command is in preview and under development. **`,
```
Comment (priteshbandi) [inline cmd/notation/policy/import.go:27]: This is incorrect, policy command is not under preview.
Can you please remove this statement? If you want to do it in next PR thats also fine as its an scope creep.

File: cmd/notation/policy/show.go [hunk 3]
```diff
@@ -20,14 +20,12 @@ func showCmd() *cobra.Command {
 		Short: "Show trust policy configuration",
 		Long: `Show trust policy configuration.
 
-** This command is in preview and under development. **
+** This command is in preview and under development. **`,
```
Comment (priteshbandi) [inline cmd/notation/policy/show.go:23]: ditto

**Review by priteshbandi — APPROVED** [review]
LGTM

File: cmd/notation/main.go [hunk 4]
```diff
@@ -30,4 +34,30 @@ func main() {
 	if err := cmd.Execute(); err != nil {
 		os.Exit(1)
 	}
+
+	// generates MD CLI docs if the GEN_CLI_DOCS env var is "true"
+	genDocs := os.Getenv("GEN_CLI_DOCS")
+	if genDocs == "true" {
```
Comment (priteshbandi) [inline cmd/notation/main.go:40]: usually we use 0 or 1 values for flags.

File: cmd/notation/main.go [hunk 5]
```diff
@@ -30,4 +34,30 @@ func main() {
 	if err := cmd.Execute(); err != nil {
 		os.Exit(1)
 	}
+
+	// generates MD CLI docs if the GEN_CLI_DOCS env var is "true"
+	genDocs := os.Getenv("GEN_CLI_DOCS")
```
Comment (priteshbandi) [inline cmd/notation/main.go:39]: Instead of doing it here should we create a new command to generate markdowns like [clockify](https://github.com/daachi/clockify-cli/blob/77d97d9537fcb311735989bed62fe969995dc3e3/cmd/gendocs.go#L39-L86]? Its much cleaner and allows you to generate markdown files to given destination

Comment (zr-msft) [reply cmd/notation/main.go:39]: @priteshbandi I'm open to discussing this approach, but I'd also like @FeynmanZhou  and @yizha1 to weigh in if they have a preference.

File: cmd/notation/cert/add.go [hunk 6]
```diff
@@ -28,14 +28,12 @@ func certAddCommand(opts *certAddOpts) *cobra.Command {
 			opts.path = args
 			return nil
 		},
-		Long: `Add certificates to the trust store
+		Long: `Add certificates to the trust store.`,
```
Comment (shizhMSFT) [inline cmd/notation/cert/add.go:31]: Should we make all `Short` and `Long` consistent by removing the suffix dot `.`?

File: cmd/notation/inspect.go [hunk 7]
```diff
@@ -60,25 +60,25 @@ func inspectCommand(opts *inspectOpts) *cobra.Command {
...
-Example - [Experimental] Inspect signatures on an OCI artifact identified by a digest using the Referrers API, if not supported (returns 404), fallback to the Referrers tag schema
-  notation inspect --allow-referrers-api <registry>/<repository>@<digest>
+# [Experimental] Inspect signatures on an OCI artifact identified by a digest using the Referrers API, if not supported (returns 404), fallback to the Referrers tag schema
+notation inspect --allow-referrers-api <registry>/<repository>@<digest>
 `
 	command := &cobra.Command{
 		Use:   "inspect [reference]",
 		Short: "Inspect all signatures associated with the signed artifact",
 		Long:  longMessage,
+		Example: exampleMessage + experimentalExamples,
```
Comment (shizhMSFT) [inline cmd/notation/inspect.go:81]: This seems not right as `experimentalExamples` should not be added if the experimental flag is not on.

Comment (shizhMSFT) [reply cmd/notation/inspect.go:81]: This also applies to other occurrences.

Comment (zr-msft) [reply cmd/notation/inspect.go]: fixed

File: cmd/notation/main.go
[the remark this answers is not in the document]
Comment (zr-msft) [reply cmd/notation/main.go:1]: fixed

File: cmd/notation/cert/add.go (hunk 6 above)
Comment (zr-msft) [reply cmd/notation/cert/add.go]: fixed

File: cmd/notation/main.go (hunk 5 above)
Comment (zr-msft) [reply cmd/notation/main.go]: thank you @priteshbandi 

I've refactored to incorporate the approach you mentioned.

File: cmd/notation/main.go (hunk 4 above)
Comment (zr-msft) [reply cmd/notation/main.go]: addressed by refactor

**Comment by rayoef** [comment]
@yizha1 Besides the private preview comments, is this PR OK to merge?

**Comment by yizha1** [comment]
Hi @rayoef  It seems this PR is no longer up-to-date. @FeynmanZhou Could you please advise on the next steps for this PR, since it affects the CLI doc generation?

**Comment by yizha1** [comment]
Close this PR. @zr-msft is being honored in memoriam.
